HelpSource/scdoc.js

```diff
@@ -105,7 +105,7 @@ function addInheritedMethods() {
             var m = mets[j];
             if(doc.methods.indexOf("_"+m.slice(1))<0) { // ignore methods already documented in this helpfile
                 var li = document.createElement("li");
-                li.innerHTML = "<a href='"+helpRoot+"/"+s.path+".html#"+m.slice(1)+"'>"+m.slice(2)+"</a>";
+                li.innerHTML = "<a href='"+helpRoot+"/"+s.path+".html#"+encodeURIComponent(m.slice(1))+"'>"+m.slice(2)+"</a>";
```

Comment:
OK I did test it with full urls with anchors query strings and spaces and
it worked correctly. .# should not be escaped, it is a valid url character.
It specifies the anchor A tag on the page.

  Do you mean escaping a second # after the first? That woyld be an invalid
anchor name. There is no reason to have the # character in am anchor name.

On Sat, Aug 20, 2016, 20:15 Nathan Ho <notifications@github.com> wrote:

> In HelpSource/scdoc.js
> <https://github.com/supercollider/supercollider/pull/2135#discussion_r75582699>
> :
>
> > @@ -105,7 +105,7 @@ function addInheritedMethods() {
> >              var m = mets[j];
> >              if(doc.methods.indexOf("_"+m.slice(1))<0) { // ignore methods already documented in this helpfile
> >                  var li = document.createElement("li");
> > -                li.innerHTML = "<a href='"+helpRoot+"/"+s.path+".html#"+m.slice(1)+"'>"+m.slice(2)+"</a>";
> > +                li.innerHTML = "<a href='"+helpRoot+"/"+s.path+".html#"+encodeURIComponent(m.slice(1))+"'>"+m.slice(2)+"</a>";
>
> @crucialfelix <https://github.com/crucialfelix> Tom is correct, but the
> point about helpRoot and s.path is also valid. those might contain unsafe
> characters.
>
> —
> You are receiving this because you were mentioned.
>
>
> Reply to this email directly, view it on GitHub
> <https://github.com/supercollider/supercollider/pull/2135/files/cd3d263606a632451b61cabc960489ac77076970#r75582699>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AANWcorRAF8wfpb0rwt6TAroveomp26bks5qh0QrgaJpZM4IpIS5>
> .
>
-- 
https://twitter.com/crucialfelix
https://medium.com/@crucialfelix
http://www.mattermind.com/